CMakeLists.txt

```diff
@@ -31,6 +23,12 @@ message(STATUS "VMA_DEBUG_INITIALIZE_ALLOCATIONS = ${VMA_DEBUG_INITIALIZE_ALLOCA
 message(STATUS "VMA_DEBUG_GLOBAL_MUTEX = ${VMA_DEBUG_GLOBAL_MUTEX}")
 message(STATUS "VMA_DEBUG_DONT_EXCEED_MAX_MEMORY_ALLOCATION_COUNT = ${VMA_DEBUG_DONT_EXCEED_MAX_MEMORY_ALLOCATION_COUNT}")
 
+if(VMA_STATIC_VULKAN_FUNCTIONS OR NOT TARGET Vulkan::Headers)
+    find_package(Vulkan REQUIRED)
+else()
+    get_target_property(Vulkan_INCLUDE_DIRS Vulkan::Headers INTERFACE_INCLUDE_DIRECTORIES)
+endif()
+
```

Comment:
Seems simpler to move this code to `src/CMakeLists.txt`

Reply:
Why? It's better if a missing required dependency halts the build sooner. And imported targets found in the top CMakeLists.txt make them available in subdirectories if necessary.

Reply:
You can't merge both codes, the condition is not exactly the same in each file :p
The linking logic is different from the finding logic.
This is what the correct merge would look like (note the double `find_package`):
```
if(VMA_STATIC_VULKAN_FUNCTIONS)
    find_package(Vulkan REQUIRED)
    target_link_libraries(VulkanMemoryAllocator PUBLIC Vulkan::Vulkan)
else()
    if (TARGET Vulkan::Headers)
        get_target_property(Vulkan_INCLUDE_DIRS Vulkan::Headers INTERFACE_INCLUDE_DIRECTORIES)
    else()
        find_package(Vulkan REQUIRED)
    endif()
    target_include_directories(VulkanMemoryAllocator PUBLIC "${Vulkan_INCLUDE_DIRS}")
endif()
```
Also if you need to link with the Vulkan library in multiple subdirectories you will do this from the top `CMakeLists.txt`:
```
find_package(Vulkan REQUIRED)
add_subdirectory(src)
add_subdirectory(test)
...
```
And not execute `find_package` in each subdirectory.